src/Decompiler/Analysis/SsaTransform.cs

```diff
@@ -385,6 +387,14 @@ private void GenerateUseDefsForUnknownCallee(CallInstruction ci)
             }
         }
 
+        private bool IsTrashed(Storage stg)
+        {
+            if (!(stg is RegisterStorage) || stg is TemporaryStorage)
+                return false;
+            return program.Platform.CreateTrashedRegisters()
+                .Where(r => r.OverlapsWith(stg)).Any();
+        }
```

Comment:
Calling `Platform.CreateTrashedRegisters()` is only needed once; you could do that outside of this functoin and pass it in as a parameter.